Sources_v3/Controllers/MemberController/MemberController_Tests.swift

```diff
@@ -0,0 +1,471 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+ import CoreData
+ @testable import StreamChatClient
+ import XCTest
+
+ final class MemberController_Tests: StressTestCase {
+    fileprivate var env: TestEnvironment!
+
+    var userId: UserId!
+    var cid: ChannelId!
+    var client: ChatClient!
+    var controller: ChatMemberController!
+    var controllerCallbackQueueID: UUID!
+    /// Workaround for uwrapping **controllerCallbackQueueID!** in each closure that captures it
+    private var callbackQueueID: UUID { controllerCallbackQueueID }
+
+    override func setUp() {
+        super.setUp()
+
+        env = TestEnvironment()
+        client = _ChatClient.mock
+        userId = .unique
+        cid = .unique
+        controller = ChatMemberController(userId: userId, cid: cid, client: client, environment: env.environment)
+        controllerCallbackQueueID = UUID()
+        controller.callbackQueue = .testQueue(withId: controllerCallbackQueueID)
+    }
+
+    override func tearDown() {
+        userId = nil
+        cid = nil
+
+        AssertAsync {
+            Assert.canBeReleased(&controller)
+            Assert.canBeReleased(&client)
+            Assert.canBeReleased(&env)
+        }
+
+        super.tearDown()
+    }
+
+    // MARK: - Controller setup
+
+    func test_client_createsUserControllerCorrectly() throws {
+        let controller = client.memberController(userId: userId, in: cid)
+
+        // Assert `state` is correct.
+        XCTAssertEqual(controller.state, .initialized)
+
+        // Assert `client` is assigned correctly.
+        XCTAssertTrue(controller.client === client)
+
+        // Assert `userId` is correct.
+        XCTAssertEqual(controller.userId, userId)
+
+        // Assert `cid` is correct.
+        XCTAssertEqual(controller.cid, cid)
+    }
+
+    func test_initialState() throws {
+        // Assert `state` is correct.
+        XCTAssertEqual(controller.state, .initialized)
+
+        // Assert `client` is assigned correctly.
+        XCTAssertTrue(controller.client === client)
+
+        // Assert `userId` is correct.
+        XCTAssertEqual(controller.userId, userId)
+
+        // Assert `cid` is correct.
+        XCTAssertEqual(controller.cid, cid)
+    }
+
+    // MARK: - Synchronize tests
+
+    func test_synchronize_changesStateand_and_callsCompletionOnCallbackQueue() {
```

Comment:
nit: The test name has some typos

Reply:
fixed in 973619a02a94368cdd60541b66007f8e1ec5a3fe